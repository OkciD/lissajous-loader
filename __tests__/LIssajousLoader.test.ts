// @ts-ignore
import canvasSerializer from 'jest-canvas-snapshot-serializer';

expect.addSnapshotSerializer(canvasSerializer);

let requestAnimationFrameMock: jest.SpyInstance;
let cancelAnimationFrameMock: jest.SpyInstance;

jest.useFakeTimers();

describe('LissajousLoader', () => {
	beforeEach(() => {
		requestAnimationFrameMock = jest.spyOn(window, 'requestAnimationFrame')
			.mockImplementation((callback: FrameRequestCallback) => {
				return setTimeout(callback, 16);
			});

		cancelAnimationFrameMock = jest.spyOn(window, 'cancelAnimationFrame')
			.mockImplementation((requestId: number) => {
				return clearTimeout(requestId);
			});
	});

	it('raf test 1', () => {
		let lol = 4

		window.requestAnimationFrame(() => {
			lol += 5;
		});

		jest.advanceTimersByTime(5);
		expect(lol).toStrictEqual(4);

		jest.advanceTimersByTime(20);
		expect(lol).toStrictEqual(9);
	});

	it('raf test 2', () => {
		let lol = 4

		const requestId = window.requestAnimationFrame(() => {
			lol += 5;
		});
		jest.advanceTimersByTime(5);
		window.cancelAnimationFrame(requestId);

		jest.runAllTimers();
		expect(lol).toStrictEqual(4);
	});

	it('canvas test', () => {
		const canvas = document.createElement('canvas');

		canvas.width = 50;
		canvas.height = 50;

		const context = canvas.getContext('2d')!;
		context.moveTo(0, 0);
		context.lineTo(20, 20);
		context.stroke();

		expect(canvas).toMatchSnapshot();
	});

	afterEach(() => {
		requestAnimationFrameMock.mockRestore();
		cancelAnimationFrameMock.mockRestore();
	});
});
