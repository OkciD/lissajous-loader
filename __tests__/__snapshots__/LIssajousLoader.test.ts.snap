// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LissajousLoader should clear a figure after full cycle (+ pause) 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAYAAAA6/NlyAAAABmJLR0QA/wD/AP+gvaeTAAAAJElEQVRoge3BMQEAAADCoPVP7WkJoAAAAAAAAAAAAAAAAAAAbjh8AAFte11jAAAAAElFTkSuQmCC"
  height="60"
  width="60"
/>
`;

exports[`LissajousLoader should render nothing without start() called 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAYAAAA6/NlyAAAABmJLR0QA/wD/AP+gvaeTAAAAJElEQVRoge3BMQEAAADCoPVP7WkJoAAAAAAAAAAAAAAAAAAAbjh8AAFte11jAAAAAElFTkSuQmCC"
  height="60"
  width="60"
/>
`;

exports[`LissajousLoader should render the whole figure after half a cycle 1`] = `
<canvas
  data-snapshot-image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADwAAAA8CAYAAAA6/NlyAAAABmJLR0QA/wD/AP+gvaeTAAAEJklEQVRoge3aW4hVZRTA8Z/jhFKOYzlOUVONTGZJahehXrqRGUH1Zj0kRQ+VRg/RvV6CIBGrlwIhu5hIQdkFIiooKwjLihKnK5iopQ5eovIhhGa0h/Udtel4ztn77Bmsff4vM+z9rb3Wd771rbW+Cy1atGjRosVRw5gMbSfgIvSg+zDZvRjALmzE7iINTLp6cTqOx6TDdO/GJmzA7418rJEOn4EluApfYQt24kB6PxEn40ScmZ7/kNp+jnXY3IgxOA2XYA7OxWwMJp1b8avo2IFk+xRMw0yswd2pXW5Owi94AMc2KDMFl+FevI5tyeBnMV+MUoVOXI8XxY8ygFeS7Fx0NahzYrJxRwaZqtyKVc18IDEdd+It/IH38H76/x0sSm2aZRXuqNWgvc4HxmKoAEM2ig4NCXfcm/7ux5/CA34qQM+gJu3tFfP1hJzyk3G/cNfPcItw4wodwovW4efUtlM+OrEHp+aUP8hTeC6jTDeWiiCzQgSheszGSmH0EyIQZmEFlmWUqUoHvhXzrB6T8Ljo6NMihWWlB0+Kji/W2IjfJjLDcTn0VaVPRMAbjvC+TbjmAJ6RfXSq0YPnxZS63ZFT6HyRCfoK0PkPzsF23Djs+XSRbz/B+UUrTXrXpu+fPezdTWIgZo2AXjBDJPYHxS++SFRYC2Wr2rLSJtLNLoem1n0iup81gnoR7vo1fsQXoroaLfqwXsSU9XLEiLYcSgexT+TwSh4dLfYlfe34K4/urB3uwsf4QIzsG2KU52ZVnIMr8SXeFnP5Q3wkUuCI0CE699iw55eLenux+pVbHtrxaNJx6bB3j4jp1Widn4nVWK56cJoiauK1mFqgzqnpm++K1Vg1VuKlAnWCBfgG42q0GSOWZ7twc0E6d+IutTPAeLEULUInYum1E+c12H4m+oVHTM6hryvJ9ms8v84RxceEHPr+xT14OaPMOFEPb8e1GeSuSTJL1famaqzEwxllqrIFF+SUvVgs+15QuybuTG02JZk8XIjvcsoepE9Ex2aYIFYxWzGvyvt56d0yzbnkGLEMndHENywQWy5FcIUwaLmICx1iobFZcXl8tVhMHJF6hUeXKM6LYI0IQvtFQOpPz2eLQqYItqlTbuYpLf/T1OvwHsWsbQmX7k86ZzmUcjYozqV7xCjnpnRBi5KlJUpYeJSutKRki4cKpVoeUsINAGJ+fZ8MaBNHIzuM3hbPDjwkvGyJyOEjtsVToRufinSzTr7ThbycIly8smOa+cwrT2nZLgLFYJIfMXeqwvikbxDHjIbuUm3El+qopVSHaaU7Li3VgXivkl15WCiuFDVLG67Dm+Ke1atir+w3vIarxQWaZlkh5ntNQ2oxVJAh08Sls7Ei+EwUXlPJ473iAlyztGvS3v/dxbTSXT0s3eXSFi1atGjR4ijibysjQfshbx4LAAAAAElFTkSuQmCC"
  height="60"
  width="60"
/>
`;
